cmd/srcd/cmd/gitbase_back_comp_test.go

```diff
@@ -0,0 +1,312 @@
+	}
+
+	return gitbaseImage[0]
+}
+
+func (s *GitbaseBackCompTestSuite) SetupTest() {
+	testDir, err := ioutil.TempDir("", "gitbase-back-comp-test")
+	s.testDir = testDir
+	if err != nil {
+		s.T().Fatalf("%v", err)
+	}
+
+	homedir, err := homedir.Dir()
+	if err != nil {
+		s.T().Fatalf("unable to get home dir: %v", err)
+	}
+
+	s.testDataDir = filepath.ToSlash(filepath.Join(homedir, ".srcd", "gitbase"))
+	os.RemoveAll(s.testDataDir)
+
+	for i := range [3]int{} {
+		s.createRepo(fmt.Sprintf("repo-%d", i))
+	}
+}
+
+func (s *GitbaseBackCompTestSuite) TearDownTest() {
+	s.CurrCmd.RunStop(context.Background())
+	s.PrevCmd.RunStop(context.Background())
+	os.RemoveAll(s.testDir)
+	os.RemoveAll(s.testDataDir)
+}
+
+func (s *GitbaseBackCompTestSuite) runInit(cmd *cmdtest.Commander, workdir string) (*bytes.Buffer, error) {
+	s.T().Helper()
+
+	ctx, cancel := context.WithTimeout(context.Background(), s.timeout)
+	defer cancel()
+
+	return cmd.RunInitWithConfig(ctx, workdir, "integration-testing-config.yaml")
+}
+
+func (s *GitbaseBackCompTestSuite) runSQL(cmd *cmdtest.Commander, query string) (*bytes.Buffer, error) {
+	s.T().Helper()
+
+	ctx, cancel := context.WithTimeout(context.Background(), s.timeout)
+	defer cancel()
+
+	return cmd.RunSQL(ctx, query)
+}
+
+func (s *GitbaseBackCompTestSuite) TestRetroCompatibleIndexes() {
+	require := s.Require()
+
+	// [previous version] srcd init s.testDir
+	_, err := s.runInit(s.PrevCmd, s.testDir)
+	require.NoError(err)
+
+	// [previous version] srcd sql "select * from repositories"
+	buf, err := s.runSQL(s.PrevCmd, "select * from repositories")
+	require.NoError(err)
+
+	expected := `+---------------+
+| REPOSITORY ID |
++---------------+
+| repo-0        |
+| repo-1        |
+| repo-2        |
++---------------+`
+	require.NoError(cmdtest.AreSQLOutputEqual(buf.String(), expected))
+
+	// [previous version] srcd sql "SHOW INDEXES"
+	buf, err = s.runSQL(s.PrevCmd, "SHOW INDEX FROM repositories")
+
+	require.NoError(err)
+	expected = `+-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+
+| TABLE | NON UNIQUE | KEY NAME | SEQ IN INDEX | COLUMN NAME | COLLATION | CARDINALITY | SUB PART | PACKED | NULL | INDEX TYPE | COMMENT | INDEX COMMENT | VISIBLE | EXPRESSION |
++-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+
++-------+------------+----------+--------------+-------------+-----------+-------------+----------+--------+------+------------+---------+---------------+---------+------------+
+`
+	require.NoError(cmdtest.AreSQLOutputEqual(buf.String(), expected))
+
+	// [previous version] srcd sql "CREATE INDEXES"
+	_, err = s.runSQL(s.PrevCmd, "CREATE INDEX repo_idx ON repositories USING pilosa (repository_id)")
+	require.NoError(err)
+	// wait a bit for index to be ready
```

Comment:
In this simple case (creating an index for 3 empty repos) I actually think that it's ok, but I totally agree that having a more formal way is better. If you think this to be a blocker I'm gonna take a look at it, otherwise I'd keep this as-is and eventually open an issue. I don't know if we create some index also somewhere else.

Reply:
maybe not a blocker for this PR, but I'd open an issue to do it at some point because I see the value of doing these tests with more real cases, where indexes will take more than 1sec to be ready :smile:

Reply:
I've created [this issue](https://github.com/src-d/engine/issues/378).